src/data/danish-gigaword.py

```diff
@@ -0,0 +1,237 @@
+                subpath = os.path.join(dagw_sektion_path, p)
+                filepaths[p] = [
+                    os.path.join(subpath, p)
+                    for p in os.listdir(subpath)
+                    if p != "LICENSE"
+                    and not p.endswith(".json")
+                    and not p.endswith(".jsonl")
+                    and not p.startswith("README")
+                ]
+
+            def handle_subdir(section):
+                return [
+                    os.path.join(filepaths[section][0], p)
+                    for p in os.listdir(filepaths[section][0])
+                    if p != "LICENSE"
+                    and not p.endswith(".json")
+                    and not p.endswith(".jsonl")
+                    and not p.startswith("README")
+                ]
+
+            filepaths["twfv19"] = handle_subdir("twfv19")
+            filepaths["datwitter"] = handle_subdir("datwitter")
+
+            return filepaths
+
+        def _read_metadata(section):
+            if section not in ["datwitter", "twfv19"]:
+                with open(os.path.join(data_file, section, section + ".jsonl")) as f:
+                    meta = ndjson.load(f)
+                if section not in ["danavis", "depbank", "botxt"]:
+                    # convert list of dicts to dict with doc_id as key
+                    try:
+                        metadict = {
+                            d["doc_id"]: {
+                                "uri": d["uri"],
+                                "date_built": d["date_built"],
+                            }
+                            for d in meta
+                        }
+                    except KeyError:
+                        # not all sections have an uri
+                        metadict = {
+                            d["doc_id"]: {
+                                "uri": "NA",
+                                "date_built": d["date_built"],
+                            }
+                            for d in meta
+                        }
+                # no uri or date_built for danavis
+                if section in ["danavis", "botxt"]:
+                    metadict = {
+                        d["doc_id"]: {
+                            "uri": "NA",
+                            "date_built": "NA",
+                        }
+                        for d in meta
+                    }
+                if section == "depbank":
+                    metadict = {
+                        d["doc_id"]: {
+                            "uri": d["uri"],
+                            "date_built": "NA",
+                        }
+                        for d in meta
+                    }
+
+                return metadict
+
+            # for twitter corpora
+            else:
+                return {
+                    "da_all_150420-260520.txt": {"uri": "NA", "date_built": "NA"},
+                    "da_fv19.txt": {"uri": "NA", "date_built": "NA"},
+                }
```

Comment:
seems odd I would instead do if section in ["datwitter...

Does this mean that there will be two NA rows? (which isn't clearly NA because it is strings?)Should it just not be included? Did we test if it was possible with an API key?